docs/window-shims-and-polyfills.md

```diff
@@ -0,0 +1,120 @@
+# Window shims and polyfills.
+
+Please consult the [gotchas](./gotchas.md), as it containes the preferred way to solve the problems where we need those shims and polyfills for.
+
+As part of the default installation we now provide a sample file on how you can provide `window` and `document` shims to your universal application. We created this on popular demand, and it serves as a starting point. 
+
+> ## Note
+> Please be aware that using the global scope can lead to memory leaks, and security issues.
+> The shims and polyfills provided will not cause this by themselves, but if your application or any of its dependencies,
+> stores or modifies _anything_ that is in the global scope (this includes, but is not limited to `window` and `document`) you are at risk.
+
+
+## Why
+
+We see a lot of questions on ngUniversal that revolve around using 3rth party libraries. With this addition we do provide a starting point, so you so have some samples and an idea how to tackle those issues. Once more, we would like to express that this is a last-resort solution, and you should really be using the other options available. The preferred way to deal with with this is utilizing the Angular injector.
```

Comment:
We see a lot of questions on ngUniversal that revolve around using 3rd party libraries. With this addition, we are providing a starting point to give you some samples and an idea on how to tackle issues related to using 3rd party libraries. We would like to emphasize that these techniques are a last-resort solution. The preferred way to deal with 3rd party libraries is by utilizing the Angular injector.